SwiftCompilerSources/Sources/SIL/Instruction.swift

```diff
@@ -353,7 +353,36 @@ final public class HopToExecutorInst : Instruction, UnaryInstruction {}
 
 final public class FixLifetimeInst : Instruction, UnaryInstruction {}
 
-final public class DebugValueInst : Instruction, UnaryInstruction {}
+// VarDecl is a struct wrapper around a C++ VarDecl pointer. This insulates the Swift interface from the C++ interface and avoids the need to fake a Swift class. When the AST exposes VarDecl, then this can be replaced whenever it is convenient.
+public struct VarDecl {
+  var bridged: BridgedVarDecl
+  
+  public init?(bridged: OptionalBridgedVarDecl) {
+    guard let decl = bridged.decl else { return nil }
+    self.bridged = BridgedVarDecl(decl: decl)
+  }
+  
+  public var userFacingName: String { String(bridged.getUserFacingName()) }
+}
+
+// See C++ VarDeclCarryingInst
+public protocol VarDeclInst {
```

Comment:
nitpick: So far we named concrete instruction classes "...Inst" and base classes/protocols "...Instruction" (with the exception of `TermInst`). Maybe this is coincidence, but IMO it makes sense, because then it's immediately clear if it's a concrete instruction or not.